### PR TITLE
Add support for Harbor Integration

### LINKED ISSUE #2023: Feature Request: Add ability to configure Harbor Project Integration
Similar to other integrations supported by this project, it would be awesome if go-gitlab could be used to setup the Harbor Project Integration. 

API Documentation: https://docs.gitlab.com/ee/api/integrations.html#harbor

### PR DESCRIPTION
**Pull request by houstonj1**
https://docs.gitlab.com/ee/api/integrations.html#harbor

Fixes #2023

### CODE REVIEW

File: services.go [hunk 1]
```diff
@@ -767,6 +767,103 @@ func (s *ServicesService) DeleteGithubService(pid interface{}, options ...Reques
...
+// HarborServiceProperties represents Harbor specific properties.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/integrations.html#harbor
+type HarborServiceProperties struct {
+	URL                  string    `json:"url"`
+	ProjectName          string    `json:"project_name"`
+	Username             string    `json:"username"`
+	Password             string    `json:"password"`
+	UseInheritedSettings BoolValue `json:"use_inherited_settings"`
```
Comment (svanharmelen) [inline services.go:788]: As far as I can tell this should be a regular `bool`

**Review by svanharmelen — CHANGES_REQUESTED** [review]
One minor comment, the rest looks good 👍🏻

File: services.go (hunk 1 above)
Comment (houstonj1) [reply services.go]: Updated! Thank you for the quick feedback.

**Review by svanharmelen — APPROVED** [review]
Thanks @houstonj1! LGTM 👍🏻

**Pull request merged**